sdk/src/plugin/proposal.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  InvalidAddressError,
+  InvalidProposalIdError,
+  OSX_COMPACT_PROPOSAL_ID_REGEX,
+  isAddress,
+  isExtendedProposalId,
+  isProposalId,
+  strip0x,
+} from '../hex';
+import {hexZeroPad} from '@ethersproject/bytes';
+
+/**
+ * Encodes the particles of a proposalId into a globally unique value for subgraph
+ *
+ * @export
+ * @param {string} pluginAddress
+ * @param {number} id
+ * @return {string}
+ */
+export function encodeProposalId(pluginAddress: string, id: number) {
+  if (!isAddress(pluginAddress)) {
+    throw new InvalidAddressError(pluginAddress);
+  }
+  return `${pluginAddress}_0x${id.toString(16)}`;
+}
+
+/**
+ * Decodes a proposalId from subgraph and returns the original pluginAddress and the nonce
+ *
+ * @export
+ * @param {string} proposalId
+ * @return {string} pluginAddress
+ * @return {number} id
+ */
+export function decodeProposalId(proposalId: string): {
+  pluginAddress: string;
+  id: number;
+} {
+  if (!isProposalId(proposalId)) {
+    throw new InvalidProposalIdError(proposalId);
+  }
+
+  const matchedRegexResult =
+    proposalId.match(OSX_COMPACT_PROPOSAL_ID_REGEX) || [];
+  if (matchedRegexResult.length !== 3) {
+    throw new InvalidProposalIdError(proposalId);
+  }
+
+  return {
+    pluginAddress: matchedRegexResult[1],
+    id: parseInt(strip0x(matchedRegexResult[2]), 16),
+  };
```

Comment:
Might be useful to rename `3` and the indicies as const values to indicate what these are

Reply:
or at least add comment explaining the indices